tern/classes/oci_image.py

```diff
@@ -0,0 +1,164 @@
+            return self.repotag
+        if self.image_id is not None:
+            return self.image_id
+        raise NameError("Image object initialized with no repotag or ID")
+
+    def get_image_manifest(self):
+        '''Assuming that there is a temp folder with a manifest.json of
+        an image inside, get a dict of the manifest.json file'''
+        temp_path = rootfs.get_working_dir()
+        with general.pushd(temp_path):
+            with open(manifest_file) as f:
+                json_obj = json.loads(f.read())
+        return json_obj
+
+    def get_image_layers(self, manifest):
+        '''Given the manifest, return the layers'''
+        layers = []
+        for layer in manifest.get('layers'):
+            layers.append(layer.get("digest").split(":")[1])
+        return layers
+
+    def get_image_config_file(self, manifest):
+        '''Given the manifest, return the config file'''
+        return manifest.get('config').get("digest").split(":")[1]
+
+    def get_image_id(self, manifest):
+        '''Given the manifest, return the image id
+        This happens to be the config file's sha256sum'''
+        config_file = self.get_image_config_file(manifest)
+        return config_file
+
+    def get_image_repotag(self):
+        '''Returns the image tag string from OCI image
+        index file'''
+        temp_path = rootfs.get_working_dir()
+        annotations = None
+        with general.pushd(temp_path):
+            with open("index.json") as f:
+                json_obj = json.loads(f.read())
+                annotations = json_obj.get("manifests")[0].get("annotations")
+        return annotations.get("org.opencontainers.image.ref.name")
+
+    def get_layer_sha(self, layer_path):
+        '''OCI's layers are file paths starting with the ID.
+        Get just the sha'''
+        return os.path.dirname(layer_path)
+
+    def get_image_config(self, manifest):
+        '''Assuming there now exists a working directory where the image
+        metadata exists, return the image config'''
+        config_file = self.get_image_config_file(manifest)
+        temp_path = rootfs.get_working_dir()
+        with general.pushd(temp_path):
+            with open(config_file) as f:
+                json_obj = json.loads(f.read())
+        return json_obj
+
+    def get_image_history(self, config):
+        '''If the config has the image history return it. Else return None'''
+        if 'history' in config.keys():
+            return config['history']
+        return None
+
+    def get_diff_ids(self, config):
+        '''Given the image config, return the filesystem diff ids'''
+        diff_ids = []
+        for item in config['rootfs']['diff_ids']:
+            diff_ids.append(item.split(':').pop())
+        return diff_ids
+
+    def get_diff_checksum_type(self, config):
+        '''Get the checksum type that was used to calculate the diff_id
+        of the image'''
+        return config['rootfs']['diff_ids'][0].split(':')[0]
```

Comment:
We already know the image checksum type using the `index.json` digest. It gets applied to all the layers.